tee-worker/ts-tests/parachain-api/prepare-build/interfaces/identity/definitions.ts

```diff
@@ -206,37 +206,73 @@ export default {
         GenericEventWithAccount: {
             account: "AccountId",
         },
-        SetUserShieldingKeyResponse: {
+        SetUserShieldingKeyResult: {
             account: "AccountId",
             id_graph: "AesOutput",
-            req_ext_hash: "H256",
         },
-        LinkIdentityResponse: {
+        LinkIdentityResult: {
             account: "AccountId",
             identity: "AesOutput",
             id_graph: "AesOutput",
-            req_ext_hash: "H256",
         },
-        DeactivateIdentityResponse: {
+        DeactivateIdentityResult: {
             account: "AccountId",
             identity: "AesOutput",
-            req_ext_hash: "H256",
         },
-        ActivateIdentityResponse: {
+        ActivateIdentityResult: {
             account: "AccountId",
             identity: "AesOutput",
-            req_ext_hash: "H256",
         },
-        SetIdentityNetworksResponse: {
-            req_ext_hash: "H256",
-        },
-        RequestVCResponse: {
+        RequestVCResult: {
             account: "AccountId",
             assertion: "Assertion",
             vc_index: "H256",
             vc_hash: "H256",
             vc_payload: "AesOutput",
+        },
+        ErrorDetail: {
+            _enum: {
+                ImportError: "Null",
+                UnauthorizedSigner: "Null",
+                StfError: "(Bytes)",
+                SendStfRequestFailed: "Null",
+                UserShieldingKeyNotFound: "Null",
+                ParseError: "Null",
+                DataProviderError: "(Bytes)",
+                InvalidIdentity: "Null",
+                WrongWeb2Handle: "Null",
+                UnexpectedMessage: "Null",
+                WrongSignatureType: "Null",
+                VerifySubstrateSignatureFailed: "Null",
+                VerifyEvmSignatureFailed: "Null",
+                RecoverEvmAddressFailed: "Null",
+                Web3NetworkOutOfBounds: "Null",
+            },
+        },
+        StfError: {
+            _enum: {
+                MissingPrivileges: "(LitentryIdentity)",
+                RequireEnclaveSignerAccount: "Null",
+                Dispatch: "(String)",
+                MissingFunds: "Null",
+                InvalidNonce: "(Index, Index)",
+                StorageHashMismatch: "Null",
+                InvalidStorageDiff: "Null",
+                InvalidMetadata: "Null",
+                SetUserShieldingKeyFailed: "(ErrorDetail)",
+                LinkIdentityFailed: "(ErrorDetail)",
+                DeactivateIdentityFailed: "(ErrorDetail)",
+                ActivateIdentityFailed: "(ErrorDetail)",
+                RequestVCFailed: "(Assertion, ErrorDetail)",
+                SetScheduledMrEnclaveFailed: "Null",
+                SetIdentityNetworksFailed: "(ErrorDetail)",
+                InvalidAccount: "Null",
+                UnclassifiedError: "Null",
+            },
+        },
+        Response: {
```

Comment:
@Kailai-Wang what do you think about `TrustedOperationResult` ?

Reply:
Maybe `TrustedOperationResponse`?  Another dumb question is why we use `Vec<u8>` for `value`, not `TrustedCallResult`? It seems in code that's the only source for encoding

Reply:
I think because for some cases the `value` can hold encoded string

```
	pub fn from_error_message(error_msg: &str) -> Self {
		RpcReturnValue {
			value: error_msg.encode(),
			do_watch: false,
			status: DirectRequestStatus::Error,
		}
	}
```

Reply:
In some cases it can hold encoded string error 

```
	pub fn from_error_message(error_msg: &str) -> Self {
		RpcReturnValue {
			value: error_msg.encode(),
			do_watch: false,
			status: DirectRequestStatus::Error,
		}
	}
```

Reply:
I see, thanks. So I need to decode it to different types based on the `status` field in the `RpcReturnValue`?

Reply:
Yes, not very convenient.